src/App.tsx

```diff
@@ -14,11 +14,148 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[],
+  isStarted: boolean,
+  isReversed: boolean,
+  sortBy: string,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isStarted: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  started = () => {
+    this.setState((prevState) => ({
+      isStarted: !prevState.isStarted,
+    }));
+  };
+
+  reverseGoods = () => {
+    this.setState(prevState => ({
+      isReversed: !prevState.isReversed,
+    }));
+  };
+
+  setSortByAlphabetically = () => {
+    const isSortByAlphabetical = this.state.sortBy === 'alphabetically';
+
+    this.setState({ sortBy: isSortByAlphabetical ? '' : 'alphabetically' });
+  };
+
+  setSortByLength = () => {
+    const isSortByLength = this.state.sortBy === 'length';
+
+    this.setState({ sortBy: isSortByLength ? '' : 'length' });
+  };
+
+  defaultGoodsList = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  };
+
+  getGoodsList = (isReversed: boolean, sortBy: string) => {
+    const copyGoods = [...this.state.goods];
+
+    copyGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case 'alphabetically':
+          return good1.localeCompare(good2);
+
+        case 'length':
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      copyGoods.reverse();
+    }
+
+    return copyGoods;
+  };
+
+  render() {
+    const {
+      isStarted, isReversed, sortBy,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">Goods</h1>
+        {
+          isStarted && (
+            <>
+              <ul className="App__list">
+                {this.getGoodsList(isReversed, sortBy).map(good => {
+                  return (
+                    <li key={good} className="App__item">
+                      {good}
+                    </li>
+                  );
+                })}
```

Comment:
- This should be separate component `GoodsList`
- Avoid using `return` in map, just use `()`